packages/@aws-cdk/aws-stepfunctions-tasks/lib/codebuild/start-build.ts

```diff
@@ -109,4 +183,17 @@ export class CodeBuildStartBuild extends sfn.TaskStateBase {
       Value: environmentVariables[name].value,
     }));
   }
+
+  private validateOverridingParameters(props: CodeBuildStartBuildProps) {
+    if (props.overrides?.secondaryArtifacts && props.overrides!.secondaryArtifacts!.length > 12) {
+      throw new Error('The maximum array length of override property "secondaryArtifacts" is 12.');
+    }
+    if (props.overrides?.secondarySources && props.overrides!.secondarySources!.length > 12) {
+      throw new Error('The maximum array length of override property "secondarySources" is 12.');
+    }
+    if (props.overrides?.timeout && (props.overrides!.timeout!.toMinutes() < 5
+      || props.overrides!.timeout!.toMinutes() > 480)) {
+      throw new Error('The value of override property "timeout" must be between 5 and 480 minutes.');
+    }
```

Comment:
is this being tested?

Reply:
there is a new test case `test('Task with illegal queuedTimeoutInMinutesOverride parameter', () => {…})` testing it.